sktime/forecasting/arch.py

```diff
@@ -0,0 +1,307 @@
+#!/usr/bin/env python3 -u
+# -*- coding: utf-8 -*-
+# copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+
+
+"""Implements Generalized Autoregressive Conditional Heteroskedasticity (GARCH) models and it's variants."""
+
+__author__ = ["Vasudeva-bit"]
+
+from sktime.forecasting.base import BaseForecaster
+from sktime.utils.validation._dependencies import _check_soft_dependencies
+
+_check_soft_dependencies("arch", severity="warning")
+
+class ARCH(BaseForecaster):
+    """Generalized Autoregressive Conditional Heteroskedasticity (GARCH) model for forecasting 
+    votility in high frequency time series data.
+
+    Implements Generalized Autoregressive Conditional Heteroskedasticity (GARCH) models introduced 
+    and implemented in [1]_, [2]_, and [3]_ to forecast the volatility in high frequency time series data.
+    
+    ARCH models are a popular class of volatility models that use observed values of returns or residuals 
+    as volatility shocks. A basic GARCH model is specified as
+        rt = μ+ϵt
+        ϵt = σtet
+        σ^2t = ω+αϵ^2t−1+βσ^2t−1
+    
+    A complete ARCH model is divided into three components:
+        a mean model, e.g., a constant mean or an ARX;
+        a volatility process, e.g., a GARCH or an EGARCH process; and
+        a distribution for the standardized residuals.
+
+    Hyper-parameters
+    ----------------
+    y : {np.ndarray, Series, None}
```

Comment:
y and x are not hyperparameters

Also, kindly adhere to numpydoc, this should be called "Parameters"

Reply:
Yeah, I shouldn't have overlooked this part.